File: code/smv-debugger/src/main/java/org/fbme/smvDebugger/panel/condition/Negation.java
package org.fbme.smvDebugger.panel.condition;

import java.util.Map;

public class Negation implements Expression {
  private final Expression expression;

  public Negation(final Expression expression) {
    this.expression = expression;
  }

  public Expression getExpression() {
    return expression;
  }

  @Override
  public boolean evaluate(final Map<String, String> itemSimpleNameToValueMap) {
    return !expression.evaluate(itemSimpleNameToValueMap);
  }

  @Override
  public String toString() {
    return "!(" + expression + ")";
  }
}

File: code/smv-debugger/src/main/kotlin/org/fbme/smvDebugger/panel/condition/Negation.kt
package org.fbme.smvDebugger.panel.condition

class Negation(val expression: Expression) : Expression {
    override fun evaluate(itemSimpleNameToValueMap: Map<String, String>): Boolean {
        return !expression.evaluate(itemSimpleNameToValueMap)
    }

    override fun toString(): String {
        return "!($expression)"
    }
}
